solver/src/driver.rs

```diff
@@ -195,10 +195,33 @@ impl Driver {
             );
             // This is an additional debug log so that the log message doesn't get too long as
             // settlement information is recoverable through tenderly anyway.
-            tracing::warn!("settlement failure for: \n{:#?}", settlement.settlement,);
+            tracing::warn!("settlement failure for: \n{:#?}", settlement,);
         }
     }
 
+    // Rate settlements, ignoring those for which the rating procedure failed.
+    async fn rate_settlements(
+        &self,
+        settlements: Vec<Settlement>,
+        prices: &HashMap<H160, BigRational>,
+    ) -> Result<Vec<RatedSettlement>> {
+        let mut result: Vec<RatedSettlement> = Vec::new();
+        for settlement in settlements {
+            let surplus = settlement.total_surplus(prices);
+            let gas_estimate = settlement_submission::estimate_gas(
+                &self.settlement_contract,
+                &settlement.clone().into(),
+            )
+            .await?;
```

Comment:
It would be nice to do this as a batched call like with the simulations. Doesn't have to be in this PR. Also not sure if ethcontract's estimateGas works with batch transport.

Reply:
I think I know what you mean and agree but I'll leave that optimization to another PR to avoid delaying this one too much.